UNRELEASED.md

```diff
@@ -2,6 +2,11 @@
 
 > Place your changes below this line.
 
+**Added:**
+
+- bpk-theming:
+  - Added `forwardedRef` prop to `withTheme` HOC, for passing down custom refs to the wrapped component.
```

Comment:
Shall we make this a breaking change? According to react docs adding forwardRef should be treated as a breaking change https://reactjs.org/docs/forwarding-refs.html#note-for-component-library-maintainers

Reply:
Yeah you might be right. Better safe than sorry anyway, I'll change it.

Reply:
Actually, I think we've got it wrong. 

The part of the document that talks about breaking changes is where it talks about using `React.forwardRef`. The part about adding a prop to HOCs - like we are - can't be breaking, as we're simply adding a new prop, it doesn't change any existing behaviour.

Reply:
Hmm, that's true, but I think we are doing things wrong then. We shouldn't be exposing the `forwardedRef` prop, but instead returning `React.forwardRef` at the end. 

```
return React.forwardRef((props, ref) => {
    return <Component {...props} forwardedRef={ref} />;
  });
```
So for people using `withTheme` the returned element will still accept `ref` which is what is expected

Reply:
Ah yeah that makes sense. Will try it

Reply:
I tried it and it broke almost all of our tests:

```
    Invariant Violation: Objects are not valid as a React child (found: object with keys {$$typeof, render}). If you meant to render a collection of children, use an array instead.
        in withTheme(BpkHorizontalNavSelectedIndicator)
        in BpkAppearanceProvider
        in Unknown
```

I feel like we should stick with the original behaviour, given that's what that page specifically recommends for HOCs?

Reply:
How do you mean? this is the example in react docs 
```js
function logProps(Component) {
  class LogProps extends React.Component {
    componentDidUpdate(prevProps) {
      console.log('old props:', prevProps);
      console.log('new props:', this.props);
    }

    render() {
      const {forwardedRef, ...rest} = this.props;

      // Assign the custom prop "forwardedRef" as a ref
      return <Component ref={forwardedRef} {...rest} />;
    }
  }

  // Note the second param "ref" provided by React.forwardRef.
  // We can pass it along to LogProps as a regular prop, e.g. "forwardedRef"
  // And it can then be attached to the Component.
  return React.forwardRef((props, ref) => {
    return <LogProps {...props} forwardedRef={ref} />;
  });
}
```

Note the return of the HOC function is `forwardRef` so consumers don't need to know for this specific component ref is passed using `forwardedRef` which is not a common practice.